resources/assets/lib/chat/conversation-view.tsx

```diff
@@ -45,8 +59,32 @@ export default class ConversationView extends React.Component<any, any> {
       return;
     }
 
-    if (this.props.dataStore.uiState.chat.autoScroll) {
-      $(chatView).scrollTop(chatView.scrollHeight);
+    const dataStore = this.props.dataStore;
+    const channel = dataStore.channelStore.channels.get(dataStore.uiState.chat.selected);
+    if (!channel.loaded) {
+      return;
+    }
+
+    const unreadMarker = $(chatView).find(`.${this.baseCssClass}__${this.unreadMarkerClass}`);
+    this.unreadMarker = unreadMarker.length > 0 ? unreadMarker : undefined;
+
+    if (this.didSwitchChannel) {
+      if (this.unreadMarker) {
+        this.scrollToUnread();
+      } else {
+        this.scrollToBottom();
+      }
```

Comment:
I wonder if this check is better done in `scrollToUnread`.

Reply:
Not sure, I figured it'd be easier to follow with this logic at a higher level? (in `componentDidUpdate`)